### ISSUE TITLE
concat: unexpected type string in list of type list in: Unexpected EOF

### ISSUE THREAD

**Issue by TechnicalMercenary** [hashicorp/terraform#19743]
### Terraform Version

```
Terraform v0.11.10
```

### Terraform Configuration Files

```hcl
data "aws_vpcs" "found" {

  count = "${local.find-vpc}"

  tags = {
    role-id = "auto-eks"
  }
}

local {


  new-vpc-id = "${element(concat(data.aws_vpcs.found.*.ids, list("")), 0)}"
}
```

### Debug Output

https://gist.github.com/TechnicalMercenary/93dede50434e7fcbd7ebc987104e6565

### Crash Output

https://gist.github.com/TechnicalMercenary/fcb4149719aac72a1483739aedaad23a

**Comment by apparentlymart** [comment]
Hi @TechnicalMercenary! Sorry Terraform crashed here.

Without seeing your full configuration it's hard to be sure, but we've seen this problem before on 0.11.10 and the problem was using a list where a string was expected on a resource argument. There are some missing validation checks that cause this to be a crash rather than a proper error in this version. The problem is fixed in master ready to be included in the forthcoming v0.12.0 release.

In this case I think the problem is that `aws_vpcs` produces a _list_ of VPCs and yet you are also using `count`, and so `data.aws_vpcs.found.*.ids` yields a list of lists rather than a list of strings as you seem to be expecting. If you just want to find all of the VPCs tagged `role-id = "auto-eks"` then you can remove the `count` argument and adjust the local value expression:

```hcl
  new-vpc-id = "${element(concat(data.aws_vpcs.found.ids, list("")), 0)}"
```

**Comment by mildwonkey** [comment]
Hi! Since there is a workaround and this is fixed in 0.12, I will close this issue, but please comment here if you still have an issue.
